### PR TITLE
PLANET-7611: Use the block name "Happy Point" consistently

### PR DESCRIPTION
**Pull request by mardelnet**
Ref: https://jira.greenpeace.org/browse/PLANET-7611

<hr>

**DESCRIPTION:**
As different names for the "Happypoint" block are used in our code (in some parts, we call it "Happypoint", in others, "happy point", "Happy Point", etc.), this PR replaces them all with the string "Happy Point", to keep it consistent.

<hr>

**RELATED PRs:**
- https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1241
- https://github.com/greenpeace/planet4-plugin-gutenberg-blocks/pull/1246

### CODE REVIEW

**Comment by planet-4** [comment]
### Test instance is ready :rocket:

:new_moon: [jupiter](https://www-dev.greenpeace.org/test-jupiter) | [admin](https://www-dev.greenpeace.org/test-jupiter/wp-admin/) | [blocks report](https://www-dev.greenpeace.org/test-jupiter/wp-admin/?page=plugin_blocks_report) | [CircleCI](https://app.circleci.com/pipelines/github/greenpeace/planet4-test-jupiter) | [composer-local.json](https://github.com/greenpeace/planet4-test-jupiter/blob/main/composer-local.json)

:watch: 2024.09.17 09:27:23

**Review by mleray — COMMENTED** [review]
I'm wondering if within this PR we could also fix the case for our Happy Point files 🤔 For example `assets/src/blocks/Happypoint/HappypointBlock.js` => `assets/src/blocks/HappyPoint/HappyPointBlock.js`
But that would require many more changes and might be overkill, it's not a big deal so let me know what you think!

**Comment by mardelnet** [comment]
> I'm wondering if within this PR we could also fix the case for our Happy Point files 🤔 For example `assets/src/blocks/Happypoint/HappypointBlock.js` => `assets/src/blocks/HappyPoint/HappyPointBlock.js` But that would require many more changes and might be overkill, it's not a big deal so let me know what you think!

Done, @mleray

**Comment by mleray** [comment]
I see one commit in this PR is about the Submenu block, that must be a mistake? Maybe needs rebasing?

Screenshot 2024-09-18 at 10 26 09

**Comment by mardelnet** [comment]
> I see one commit in this PR is about the Submenu block, that must be a mistake? Maybe needs rebasing?

I think that's because this PR was originally based on the changes in the Submenu block.
Then, those changes were reverted [here](https://github.com/greenpeace/planet4-master-theme/commit/26f889b84d08a840971697a619bf8696213d781e).
If this is too confusing, I can start a fresh PR from scratch for the Happy Point block.

**Comment by mleray** [comment]
> @mleray, I think that's because this PR was originally based on the changes in the Submenu block. Then, those changes were reverted [here](https://github.com/greenpeace/planet4-master-theme/commit/26f889b84d08a840971697a619bf8696213d781e). If this is too confusing, I can start a fresh PR from scratch for the Happy Point block.

@mardelnet if you rebase on local you can also decide to remove some commits, would that solve the issue? Otherwise it's not a big deal since we squash all commits when merging anyways.

File: assets/src/blocks/HappyPoint/HappyPointBlock.js [hunk 1]
```diff
@@ -1,14 +1,14 @@
-import {HappypointEditor} from './HappypointEditor';
-import {HappypointBlock as HappypointBlockV1} from './deprecated/HappypointBlocklV1.js';
+import {HappyPointEditor} from './HappyPointEditor.js';
+import {HappyPointBlock as HappyPointBlocklV1} from './deprecated/HappyPointBlocklV1.js';
```
Comment (mleray) [inline assets/src/blocks/HappyPoint/HappyPointBlock.js:2]: There is a typo in this file name: `HappyPointBlocklV1` => `HappyPointBlockV1`

File: assets/src/scss/editorStyle.scss [hunk 2]
```diff
@@ -41,7 +41,7 @@
 @import "blocks/Columns/ColumnsEditorStyle";
 @import "blocks/Cookies/CookiesEditorStyle";
 @import "blocks/ENForm/ENFormEditorStyle";
-@import "blocks/Happypoint/HappypointEditorStyle";
+@import "blocks/HappyPoint/HappypointEditorStyle";
```
Comment (mleray) [inline assets/src/scss/editorStyle.scss:44]: The style file's case should also be fixed: `HappypointEditorStyle` => `HappyPointEditorStyle`

File: assets/src/scss/blocks.scss [hunk 3]
```diff
@@ -9,7 +9,7 @@
 @import "blocks/Cookies/CookiesStyle";
 @import "blocks/Counter/CounterStyle";
 @import "blocks/ENForm/ENFormStyle";
-@import "blocks/Happypoint/HappypointStyle";
+@import "blocks/HappyPoint/HappypointStyle";
```
Comment (mleray) [inline assets/src/scss/blocks.scss:12]: The style file's case should also be fixed: HappypointStyle => HappyPointStyle

Comment (mardelnet) [reply assets/src/scss/blocks.scss]: Fixed!

File: assets/src/blocks/HappyPoint/HappyPointBlock.js (hunk 1 above)
Comment (mardelnet) [reply assets/src/blocks/HappyPoint/HappyPointBlock.js]: Fixed!

**Review by mleray — APPROVED** [review]
Looks good to me, good job 👏

**Pull request merged**